src/commands/setup_macos_executor.yml

```diff
@@ -27,7 +35,12 @@ steps:
         source $BASH_ENV
 
   - when:
-      condition: <<parameters.homebrew_cache>>
+      condition:
+        and:
+          - <<parameters.homebrew_cache>>
+          - or:
+              - <<parameters.android>>
+              - <<parameters.detox>>
```

Comment:
why are you adding this? If you expose the `homebrew_cache` to the user, that's the only key you should honor for the sake of the cache

Reply:
If both android and detox are false, brew is never used, hence skipping both the update and the cache - seems silly to nonetheless do caching.

Does the same logic apply to homebrew_update - should brew update run even if detox and android are disabled?

Reply:
I would just always keep brew update and allow the users to skip the `homebrew_cache` as they see fit. If you wish you can add another parameter for the `homebrew_update` step

Reply:
There already is a parameter for the homebrew update step.

In my own usage, once android and detox were gone, the brew update was adding extra time to the build for no benefit, so turning it off automatically in the orb seemed like the better approach.

Reply:
Oh you're right. Still I won't turn automatically steps and will just give the user full control